tests/end-to-end/director/application_api_test.go

```diff
@@ -257,6 +257,108 @@ func TestCreateApplicationWithDocuments(t *testing.T) {
 	assertApplication(t, in, actualApp)
 }
 
+func TestAddDependentObjectsWhenAppDoesNotExist(t *testing.T) {
+	applicationId := "foo"
+
+	t.Run("add Webhook", func(t *testing.T) {
+		//GIVEN
+		ctx := context.Background()
+		webhookInStr, err := tc.graphqlizer.WebhookInputToGQL(&graphql.WebhookInput{
+			URL:  "new-webhook",
+			Type: graphql.ApplicationWebhookTypeConfigurationChanged,
+		})
+		tmpWebhook := graphql.Webhook{}
+		require.NoError(t, err)
+
+		//WHEN
+		addReq := gcli.NewRequest(
+			fmt.Sprintf(`mutation {
+			result: addWebhook(applicationID: "%s", in: %s) {
+					%s
+				}
+			}`, applicationId, webhookInStr, tc.gqlFieldsProvider.ForWebhooks()))
+		err = tc.RunQuery(ctx, addReq, &tmpWebhook)
+
+		//THEN
+		require.Error(t, err)
```

Comment:
here you can check only if error exist, but you can be more specific and check if error represents missing application